OpenSim/Examples/BuildDynamicWalker/BuildDynamicWalkerModel.cpp

```diff
@@ -0,0 +1,351 @@
+// Make all of OpenSim and Simbody's classes available.
+#include <OpenSim/OpenSim.h>
+// "Use" the OpenSim and SimTK namespace to shorten type names.
+// Note: Several classes appear in both namespaces and require using the
+// fully-qualified class name (for example, OpenSim::Body).
+using namespace OpenSim;
+using SimTK::Vec3;
+using SimTK::Inertia;
+using SimTK::Pi;
+using SimTK::convertDegreesToRadians;
+
+// Build an OpenSim dynamic walker model and save it to an OSIM file.
+int main() {
+    try {
+        // Code to the construct the model will go here
+        // Section: Setup
+        // Define key model variables
+        double pelvisWidth = 0.20;
+        double thighLength = 0.40;
+        double shankLength = 0.435;
+
+        // Create an OpenSim Model
+        Model osimModel;
+        osimModel.setName("DynamicWalkerModel");
+
+        // Get a reference to the ground object
+        const Ground& ground = osimModel.getGround();
+
+        // Define the acceleration of gravity
+        osimModel.setGravity(Vec3(0, -9.80665, 0));
+
+        // Section: Create the Platform Body
+        double mass = 1;
+
+        // Location of the center of mass from the body origin expressed in the
+        // body frame.
+        Vec3 comLocInBody(0.0, 0.0, 0.0);
+
+        // Inertia of the body expressed in the body frame
+        Inertia bodyInertia(1.0, 1.0, 1.0, 0.0, 0.0, 0.0);
+
+        // Create the body
+        Body* platform = new Body("Platform", mass, comLocInBody, bodyInertia);
+
+        // Add geometry to display in the GUI
+        platform->attachGeometry(new Brick(Vec3(1, 0.05, 1)));
+
+        // Add the Platform Body to the Model
+        osimModel.addBody(platform);
+
+        // Section: Create the Platform Joint
+        // Create the joint connection the platform to the ground
+        Vec3 locationInParent(0.0, 0.0, 0.0);
+        Vec3 orientationInParent(0.0, 0.0, 0.0);
+        Vec3 locationInChild(0.0, 0.0, 0.0);
+        Vec3 orientationInChild(0.0, 0.0, 0.0);
+        PinJoint *platformToGround = new PinJoint("PlatformToGround",
```

Comment:
The MATLAB version uses a WeldJoint with the child joint frame having the 10degrees of incline. Any reason to make the incline a DOF and then constrain it?

Reply:
That's how the example was written. Using a DOF makes it easy to run simulations in the GUI with different inclines. Changing this would also require rewriting a good portion of the Confluence tutorial; this pin joint is used to explain coordinates, etc.

Reply:
Let's leave it as it is.

Reply:
Makes sense to leave it then.